commands/new_function.go

```diff
@@ -81,12 +85,27 @@ the "Dockerfile" lang type in your YAML file.
 		return fmt.Errorf("you must supply a function language with the --lang flag")
 	}
 
-	PullTemplates("")
+	existingYaml := len(yamlFile) > 0
+	var stackFileName string
+	if !existingYaml {
+		// We will create a new YAML file for this function
+		stackFileName = functionName + ".yml"
+	} else {
+		stackFileName = yamlFile
+		// YAML file was passed in, so parse to see if it is valid
+		parsedServices, err := stack.ParseYAMLFile(stackFileName, "", "")
+		if err != nil || parsedServices == nil {
+			return fmt.Errorf("Specified file (" + stackFileName + ") is not valid YAML\n")
+		}
+		//TODO: Verify that services is good
+	}
 
 	if stack.IsValidTemplate(lang) == false {
 		return fmt.Errorf("%s is unavailable or not supported", lang)
 	}
 
+	PullTemplates("")
```

Comment:
The `new` will always fail with "X is unavailable or not supported" as you're validating before pulling in the templates. I assume you were testing with templates already present?